catalog-service/src/main/java/com/example/catalogservice/services/ProductService.java

```diff
@@ -151,11 +173,15 @@ public Mono<Product> saveProduct(ProductDto productDto) {
                         DuplicateKeyException.class,
                         e ->
                                 // Handle unique key constraint violation here
-                                Mono.error(new ProductAlreadyExistsException(productDto.code())));
+                                Mono.error(new ProductAlreadyExistsException(productDto.code())))
+                .map(productMapper::toProductResponse);
     }
 
-    public Mono<Product> updateProduct(Product product) {
-        return productRepository.save(product);
+    public Mono<ProductResponse> updateProduct(ProductDto productDto, Long id) {
+        Product product = productMapper.toEntity(productDto);
```

Comment:
here we should use updateReference, instead of manually updating the id as we have already fetched entity.